### ISSUE TITLE
Consider add an global interceptor before render UI component with data

### ISSUE THREAD

**Issue by AlloVince** [appbaseio/reactivesearch#838]
**Issue Type:** enhancement

**Platform:** web

**Description:**

Hi, me again :P 

Also here is another useful feature in searchkit: 

In my project, data not only store in Elastic,  so I need to add some extra fields or modify some values before render components, in searchkit I could do it by overwride parent class:

```js
class MySearchkitManager extends SearchkitManager {

  setResults(results) {
    return fetch('/some_other_api')
      .then(res => res.json())
      .then(() => {
        //modify or add new fields to elastic results here
        super.setResults(results);
      });
  }
}

const searchkit = new MySearchkitManager();

// start render componets
``` 

I think it will be helpful if there has global interceptor support :)

**Comment by metagrover** [comment]
We do support `transformRequest` and `transformResponse` props on `ReactiveBase` component, so that can definitely work here. [Find docs here](https://opensource.appbase.io/reactive-manual/v3/getting-started/reactivebase.html#props).

---

I also want to understand this from developer experience standpoint. I'd really appreciate if you give this a try and provide us with any feedback on this spec - particularly, if it was easier to cook up the logic that you wished to implement, (or) if there were any specific pain-points? 😅

**Comment by AlloVince** [comment]
> We do support `transformRequest` and `transformResponse` props on `ReactiveBase` component, so that can definitely work here. [Find docs here](https://opensource.appbase.io/reactive-manual/v3/getting-started/reactivebase.html#props).
> 
> I also want to understand this from developer experience standpoint. I'd really appreciate if you give this a try and provide us with any feedback on this spec - particularly, if it was easier to cook up the logic that you wished to implement, (or) if there were any specific pain-points? 😅

thanks for quick response. I tried `transformResponse` and it works, but it not very easy (or clear) to use.

In searchkit case, because all components will send only ONE request to elastic, so if I override the
response will be clear that, the response will share to all components

In reactivesearch, because components are independent, if we have 3 components, will send 3 requests and got 3 Elastic responses. each response format looks like:

```js
{
 responses: [
    {
        aggregations: {},
        hits: {
           hits: []
        }
     }
 ]
}
```

And the `transformResponse` will be triggered 3 times.

So I think the biggest problem of `transformResponse` is, I don't have a good way to finger out, current response belongs to which component.

The same problem exists in `defaultQuery`, modify query from component is not friendly, maybe I will talk about this in another issue.

However, I think a way to improve `transformRequest` / `transformResponse` is pass componentId as second param. such as

```jsx
<ReactiveBase
    transformResponse={async (response, componentId) => {
         if (componentId === 'foo') {
            //do something special to response
         }
        return response
    }}
>
```

Hope this could help

**Comment by metagrover** [comment]
Thanks for the input. I think that makes a lot of sense. We will improve this spec for better usability 👍

**Comment by metagrover** [comment]
`transformResponse` spec has been improved to avail `componentId`s with individual responses for every components. It will be released in `v3.0.0-beta.8`.

Find docs here: https://github.com/appbaseio/reactive-manual/blob/v3/content/getting-started/ReactiveBase.md#props
